### ISSUE TITLE
Drop support for python 2.7

### ISSUE THREAD

**Issue by aqeelat** [sebschrader/python-arpreq#12]
I would like to work on this

**Comment by sebschrader** [comment]
As long as it's still possible with reasonable effort to build and test manylinux wheels for the EOL versions, I don't see the point of dropping support at this time. Although Python 2.7 is EOL, there are undoubtedly still many people using it (some distros with long support cycles are still shipping and supporting Python 2.7). I'd like to avoid actively breaking compatibility without much benefit. You may submit a PR, which I'm willing to review and merge into a preparation branch for future release though.

Because this is a C extension module and not a pure Python module the differences between Python 2/Python 3/Python 3 with PEP 489 (and also between CPython and PyPy) are resolved at compile-time by the pre-processor. I tried to minimize the scope and the quantity of the pre-processor `#ifdef` sections, to keep the maintenance burden low and the readability of the code high. The current unit tests wouldn't benefit much from dropped support either.

**Comment by aqeelat** [comment]
I agree with you, but people using python 2.7 will still be able to use older versions of the library. I believe python 2.7 should be abandoned and we should encourage/persuade users to move to python 3.7+.
